src/test_application/activescenes/scenarios.cpp

```diff
@@ -0,0 +1,173 @@
+            auto &rRenderGl     = top_get< active::RenderGL >       (mainView.m_topData, idRenderGl);
+            auto &rUserInput    = top_get< input::UserInputHandler >(mainView.m_topData, idUserInput);
+
+            rActiveApp.set_on_draw(enginetest::generate_draw_func(rScene, rActiveApp, rRenderGl, rUserInput));
+        };
+    });
+
+    add_scenario("physicstest", "Physics lol",
+                 [] (MainView mainView, PkgId pkg, Sessions_t& sceneOut) -> RendererSetup_t
+    {
+        auto &rTopData = mainView.m_topData;
+        auto &rTags = mainView.m_rTags;
+        auto const idResources = mainView.m_idResources;
+        Builder_t builder{mainView.m_rTags, mainView.m_rTasks, mainView.m_rTaskData};
+
+        sceneOut.resize(9);
+        auto & [scnCommon, matVisual, physics, newton, shapeSpawn, droppers, gravity, bounds, thrower] = unpack<9>(sceneOut);
+
+        using namespace testapp::scenes;
+
+        scnCommon   = setup_common_scene    (builder, rTopData, rTags, idResources);
+        matVisual   = setup_material        (builder, rTopData, rTags, scnCommon);
+        physics     = setup_physics         (builder, rTopData, rTags, scnCommon, idResources, pkg);
+        newton      = setup_newton_physics  (builder, rTopData, rTags, scnCommon, physics);
+        shapeSpawn  = setup_shape_spawn     (builder, rTopData, rTags, scnCommon, physics, matVisual);
+        droppers    = setup_droppers        (builder, rTopData, rTags, scnCommon, shapeSpawn);
+        gravity     = setup_gravity         (builder, rTopData, rTags, scnCommon, physics, shapeSpawn);
+        bounds      = setup_bounds          (builder, rTopData, rTags, scnCommon, physics, shapeSpawn);
+
+        add_floor(rTopData, scnCommon, matVisual, shapeSpawn, idResources, pkg);
+
+        return [] (MainView mainView, Session const& magnum, Sessions_t const& scene, [[maybe_unused]] Sessions_t& rendererOut)
+        {
+            auto &rTopData = mainView.m_topData;
+            auto &rTags = mainView.m_rTags;
+            Builder_t builder{mainView.m_rTags, mainView.m_rTasks, mainView.m_rTaskData};
+
+            auto const& [scnCommon, matVisual, physics, newton, shapeSpawn, gravity, bounds, thrower] = unpack<8>(scene);
+
+            rendererOut.resize(4);
+
+            using namespace testapp::scenes;
+
+            auto & [scnRender, simpleCam, shVisual, camThrow] = unpack<4>(rendererOut);
+            scnRender   = setup_scene_renderer      (builder, rTopData, rTags, magnum, scnCommon, mainView.m_idResources);
+            simpleCam   = setup_simple_camera       (builder, rTopData, rTags, magnum, scnCommon, scnRender);
+            shVisual    = setup_shader_visualizer   (builder, rTopData, rTags, magnum, scnCommon, scnRender, matVisual);
+            camThrow    = setup_thrower             (builder, rTopData, rTags, magnum, scnRender, simpleCam, shapeSpawn);
+
+            setup_magnum_draw(mainView, magnum, scnCommon, scnRender);
+        };
+    });
```

Comment:
Important part. This is where the physics test scene is created, which is composed of Sessions created by these setup_\* functions. A new scenario can be easily created by copy-pasting this whole block of code and giving it a different name.